tst/unit/test_pararrays.cpp

```diff
@@ -0,0 +1,242 @@
+      auto view = a.Get<3>();
+      auto mirror = a.GetMirror<3>();
+      int n = 0;
+      int sum_host = 0;
+      for (int k = 0; k < N3; k++) {
+        for (int j = 0; j < N2; j++) {
+          for (int i = 0; i < N1; i++) {
+            mirror(k,j,i) = n;
+            sum_host += n;
+            n++;
+          }
+        }
+      }
+      Kokkos::deep_copy(view,mirror);
+      AND_THEN("The sum of the lower three indices is correct") {
+        int sum_device;
+        using policy = Kokkos::MDRangePolicy<Kokkos::Rank<3>>;
+        Kokkos::parallel_reduce(policy({0,0,0}, {N3,N2,N1}),
+                                KOKKOS_LAMBDA(const int k, const int j,
+                                              const int i,
+                                              int& update) {
+                                  update += a(k,j,i);
+                                },
+                                sum_device);
+        REQUIRE( sum_host == sum_device );
+      }
+      AND_THEN("the sum of the kower TWO indices is correct") {
+        int sum_host = 0;
+        int n = 0;
+        for (int j = 0; j < N2; j++) {
+          for (int i = 0; i < N1; i++) {
+            sum_host += n;
+            n++;
+          }
+        }
+        int sum_device;
+        using policy = Kokkos::MDRangePolicy<Kokkos::Rank<2>>;
+        Kokkos::parallel_reduce(policy({0,0}, {N2,N1}),
+                                KOKKOS_LAMBDA(const int j, const int i,
+                                              int& update) {
+                                  update += a(j,i);
+                                },
+                                sum_device);
+        REQUIRE( sum_host == sum_device );
+      }
+      AND_THEN("slicing is possible") {
+        auto b = a.Slice(std::make_pair(1,3),
+                         Kokkos::ALL(),Kokkos::ALL());
+        AND_THEN("The sliced array agrees with the original") {
+          int total_errors;
+          using policy = Kokkos::MDRangePolicy<Kokkos::Rank<3>>;
+          Kokkos::parallel_reduce(policy({0,0,0}, {2,N2,N1}),
+                                  KOKKOS_LAMBDA(const int k,
+                                                const int j,
+                                                const int i,
+                                                int& update) {
+                                    update += (b(k,j,i) == a(k+1,j,i)) ? 0 : 1;
+                                  },
+                                  total_errors);
+          REQUIRE( total_errors == 0 );
+        }
+      }
+    }
+  }
+}
+
+TEST_CASE("ParArrayFlat","[ParArrayFlat]") {
+  GIVEN("A ParArrayFlat with some dimensions") {
+    constexpr int N1 = 2;
+    constexpr int N2 = 3;
+    constexpr int N3 = 4;
+    ParArrayFlat<Real> a("test",N3,N2,N1);
+    THEN("It has the right size") {
+      REQUIRE( a.GetSize() == N3*N2*N1 );
+    }
+    THEN("We can get a view and use it to fill the array") {
+      auto view = a.Get(N3,N2,N1);
```

Comment:
So `ParArrayFlat` introduces similar boilerplate to what we used in K-Athena? It wasn't too cumbersome in that code.

Reply:
It does. Both `ParArrayFlat` and `ParArrayFlex` offer something like the `K-athena` syntax if you want it. For `ParArrayFlex` it looks something like
```C++
auto view = a.Get<3>();
```
The difference is in how that data is stored internally. For `ParArrayFlat`, `a.Get()` returns an unmanaged view to a block of memory it holds internally (privately, the array holds a contiguous 1D view.). For `ParArrayFlex`, `a.Get<D>()` returns the proper-dimensional view it was holding internally.